java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/common/Util.java

```diff
@@ -352,14 +352,15 @@ public static MemorySegment prepareTensor(
     return tensor;
   }
 
-  public static MemorySegment allocateRMMSegment(long resourceHandle, long datasetBytes) {
+  public static CloseableRMMAllocation allocateRMMSegment(long resourceHandle, long datasetBytes) {
```

Comment:
What about moving this to CloseableRMMAllocation? (and make the CloseableRMMAllocation ctor private?)

Reply:
Good idea.  I'll check the call-sites to make sure this is viable.

Reply:
(This change is still pending.)

Reply:
The latest commit fixes this problem.